config_example.json

```diff
@@ -15,19 +15,146 @@
         "fennec_nightly_repacks": "/path/to/beetmoverscript/beetmoverscript/templates/fennec_nightly_repacks.yml",
         "fennecx86_nightly": "/path/to/beetmoverscript/beetmoverscript/templates/fennecx86_nightly.yml"
     },
+
     "s3": {
         "firefox": {
-            "bucket": "mozilla-releng-beetmoverscript-firefox",
-            "credentials": {
-                "id": "dummy",
-                "key": "dummy"
+            "nightly": {
+                "push-to-nightly": {
+                    "s3_prefixes": {
+                        "default": {
+                            "s3_prefix": "pub/firefox/nightly/{}-{}/",
+                            "s3_prefix_latest": "pub/firefox/nightly/latest-{}/"
+                        },
+                        "repacks": {
+                            "s3_prefix": "pub/firefox/nightly/{}-{}-l10n/",
+                            "s3_prefix_latest": "pub/firefox/nightly/latest-{}-l10n/"
+                        }
+                    },
+                    "bucket": "mozilla-releng-firefox-nightly-bucket",
```

Comment:
So this is a bit different.

We were discussing having the bucket/creds be one thing (nightly vs release vs dep scopes), and behavior be another thing (push-to-nightly vs push-to-candidates etc).  This latter thing will be determined by another item in the task definition, like `task.payload.behavior` or `task.payload.beetmover.action` or something.  This could also be a second scope, similar to how signingscript allows for a [single cert type](https://github.com/mozilla-releng/signingscript/blob/master/signingscript/task.py#L25) (nightly vs dep vs release) but allows for signing formats to be specified in [another set of scopes](https://github.com/mozilla-releng/signingscript/blob/master/signingscript/task.py#L34) (gpg vs jar vs mar)

So if I were you, I'd move all the bucket/credential stuff to their own area, e.g.

```
"buckets": {
    "nightly": {  # based on project:releng:beetmover:nightly scopes
        "bucket": "...",
        "credentials": {...},
    },
    "release": {...},
    "dep": {...},
}
```

and then specify the actions / behaviors etc in their own dict, e.g.

```
"actions": {
    "push-to-nightly": {
        # path templates
     },
    "push-to-candidates": {
        # path templates
    },
```
These may someday get more than just path templates, so I'd leave as a dict rather than hardcode to a list, but up to you.  Also, depending on the action(s) in the task definition, I'd call different functions:

```
action_map = {
    'push-to-nightly': push_to_nightly_function,
    'push-to-candidates': push_to_candidates_function,
}
async def async_main(context):
    ...
    context.task = get_task(context.config)
    context.action = get_action(context.task)
    if action_map.get(context.action):
        await action_map[context.action](context)
    else:
        log.critical("Unknown action {}!".format(context.action))
        # die
```

Does this make sense?  This is what I was saying on the whiteboard, but I'm not sure if I communicated that clearly.